File: Model/Actions/ExcelExport.vb
Public Class ExcelExport
    Public fileList As String

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bSelectAll.Click
        For i As Integer = 0 To fileListBox.Items.Count - 1
            fileListBox.SetItemChecked(i, True)
        Next
    End Sub

    Private Sub bDeselectAll_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bDeselectAll.Click
        For i As Integer = 0 To fileListBox.Items.Count - 1
            fileListBox.SetItemChecked(i, False)
        Next
    End Sub

    Private Sub bOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bOK.Click
        Dim allFiles(fileListBox.Items.Count) As String
        fileList = ""
        fileListBox.Items.CopyTo(allFiles, 0)

        For i As Integer = 0 To fileListBox.Items.Count - 1
            If fileListBox.GetItemChecked(i) Then
                fileList += allFiles(i) + ","
            End If
        Next
        If fileList.Length <> 0 Then
            fileList = fileList.Substring(0, fileList.Length - 1)
        End If
        Me.Close()
    End Sub

    Private Sub bCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bCancel.Click
        Me.Close()
    End Sub
End Class

File: Model/Actions/ExcelExport.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Actions
{
    public partial class ExcelExport
    {

        public string fileList;
        private void Button1_Click(System.Object sender, System.EventArgs e)
        {
            for (int i = 0; i <= fileListBox.Items.Count - 1; i++)
            {
                fileListBox.SetItemChecked(i, true);
            }
        }

        private void bDeselectAll_Click(System.Object sender, System.EventArgs e)
        {
            for (int i = 0; i <= fileListBox.Items.Count - 1; i++)
            {
                fileListBox.SetItemChecked(i, false);
            }
        }

        private void bOK_Click(System.Object sender, System.EventArgs e)
        {
            string[] allFiles = new string[fileListBox.Items.Count + 1];
            fileList = "";
            fileListBox.Items.CopyTo(allFiles, 0);

            for (int i = 0; i <= fileListBox.Items.Count - 1; i++)
            {
                if (fileListBox.GetItemChecked(i))
                {
                    fileList += allFiles[i] + ",";
                }
            }
            if (fileList.Length != 0)
            {
                fileList = fileList.Substring(0, fileList.Length - 1);
            }
            this.Close();
        }

        private void bCancel_Click(System.Object sender, System.EventArgs e)
        {
            this.Close();
        }
        public ExcelExport()
        {
            InitializeComponent();
        }

    }
}
